mlfinlab/backtest_statistics/statistics.py

```diff
@@ -152,62 +152,96 @@ def all_bets_concentration(returns: pd.Series, frequency: str = 'M') -> tuple:
     two consecutive high-watermarks. The time under water is the time
     elapsed between an high watermark and the moment the PnL (profit and loss)
-    exceeds the previous maximum PnL.
+    exceeds the previous maximum PnL. We also append the Time under water series
+    with period from the last high-watermark to the last return observed.
 
     Return details:
-    - Drawdown series index is time of a high watermark and value of a drawdown after.
-    - Time under water index is time of a high watermark and how much time passed till
-    next high watermark in years.
+    - Drawdown series index is the time of a high watermark and the value of a
+      drawdown after it.
+    - Time under water index is the time of a high watermark and how much time
+      passed till the next high watermark in years. Also includes time between
+      the last high watermark and last observation in returns as the last element.
 
     :param returns: (pd.Series) returns from bets
-    :param dollars: (bool) flag if given dollar performance and not returns
+    :param dollars: (bool) flag if given dollar performance and not returns.
+                    If dollars, then drawdowns are in dollars, else as a %.
     :return: (tuple of pd.Series) series of drawdowns and time under water
-                                if dollars, then in dollars, else as a %
     """
     frame = returns.to_frame('pnl')
     frame['hwm'] = returns.expanding().max()  # Adding high watermarks as column
     # Grouped as min returns by high watermarks
     high_watermarks = frame.groupby('hwm').min().reset_index()
     high_watermarks.columns = ['hwm', 'min']
-    # Time high watermark occured
+    # Time high watermark occurred
     high_watermarks.index = frame['hwm'].drop_duplicates(keep='first').index
     # Picking ones that had a drawdown after high watermark
     high_watermarks = high_watermarks[high_watermarks['hwm'] > high_watermarks['min']]
     if dollars:
         drawdown = high_watermarks['hwm'] - high_watermarks['min']
     else:
         drawdown = 1 - high_watermarks['min'] / high_watermarks['hwm']
+
     time_under_water = ((high_watermarks.index[1:] -
                          high_watermarks.index[:-1]) / np.timedelta64(1, 'Y')).values
-    time_under_water = pd.Series(time_under_water, index=high_watermarks.index[:-1])
+
+    # Adding also period from last High watermark to last return observed.
+    time_under_water = np.append(time_under_water,
+                                 (returns.index[-1] - high_watermarks.index[-1]) /
+                                 np.timedelta64(1, 'Y'))
+
+    time_under_water = pd.Series(time_under_water, index=high_watermarks.index)
     return drawdown, time_under_water
 
 
-def sharpe_ratio(returns: pd.Series, cumulative: bool = False,
-                 entries_per_year: int = 252, risk_free_rate: float = 0) -> float:
+def sharpe_ratio(returns: pd.Series, entries_per_year: int = 252,
```

Comment:
You line length is squashing your code here.

Reply:
Will change